File: Web/src/org/alex73/korpus/server/data/GrammarInitial.java
package org.alex73.korpus.server.data;

import java.util.List;
import java.util.Map;
import java.util.Set;

import org.alex73.korpus.base.DBTagsGroups;
import org.alex73.korpus.base.DBTagsGroups.KeyValue;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

@JsonInclude(Include.NON_NULL)
public class GrammarInitial {
    public Map<Character, GrammarLetter> grammarTree;
    public List<KeyValue> grammarWordTypes;
    public Map<Character, DBTagsGroups> grammarWordTypesGroups;
    public Map<Character, Set<String>> skipGrammar;
    public List<GrammarDict> slouniki;
    public List<Stat> stat;
    public Map<String, Map<String, String>> localization;

    public static class Stat {
        public String title;
        public int paradigmCount;
        public int formCount;
    }

    @JsonInclude(Include.NON_NULL)
    public static class GrammarLetter {
        public String name;
        public String desc;
        public Map<Character, GrammarLetter> ch;
    }

    @JsonInclude(Include.NON_NULL)
    public static class GrammarDict {
        public String name;
        public String desc;
    }
}

File: Web/war/ts/dto/GrammarInitial.ts
class GrammarInitial {
  public grammarTree: { [key: string]: GrammarLetter; };
  public grammarWordTypes: KeyValue[];
  public grammarWordTypesGroups: { [key: string]: DBTagsGroups; };
  public skipGrammar: { [key: string]: string[]; };
  public slouniki: GrammarDict[];
  public stat: GrammarInitialStat;
}

class GrammarInitialStat {
    public title: string;
    public paradigmCount: number;
    public formCount: number;
}

class GrammarDict {
    public name: string;
    public desc: string;
}

class GrammarLetter {
  public name: string;
  public desc: string;
  public ch: { [key: string]: GrammarLetter; };
}

class KeyValue {
  public key: string;
  public value: string;
}

class DBTagsGroups {
  public groups: Group[];
}

class Group {
  public name: string;
  public formGroup: boolean;
  public items: Item[];
}

class Item {
  public code: string;
  public description: string;
}
